libs/chonky/src/lib.rs

```diff
@@ -164,15 +240,33 @@ mod tests {
         let pdf = pdf_segmentation::load_pdf(&pdfium, test_pdf_string)
             .change_context(ChonkyError::Pdfium)?;
 
+        //number of pages of pdf
+        let num_pages: usize = pdf.pages().len().into();
+
         let preprocessed_pdf =
             pdf_segmentation::pdf_to_images(&pdf).change_context(ChonkyError::Pdfium)?;
 
-        let num_pages: usize = pdf.pages().len().into(); //number of pages of pdf
-
+        //start by checking if proper amount of images are converted
         match preprocessed_pdf.len() {
-            x if x == num_pages => Ok(()),
-            _ => Err(Report::new(ChonkyError::Pdfium)
-                .attach_printable("The length of vector should be number of pages")),
+            x if x == num_pages => (),
+            _ => {
+                return Err(Report::new(ChonkyError::Pdfium)
+                    .attach_printable("The length of vector should be number of pages"));
+            }
+        }
+
+        // now check if the image contents are the same using insta snapshots
+        // start by converting images to binary
+        let preprocessed_pdf: Vec<Vec<u8>> = preprocessed_pdf
+            .into_iter()
+            .map(image::DynamicImage::into_bytes)
+            .collect();
+
+        // we only really need to check the first three pages
+        for (index, page_bytes) in preprocessed_pdf.into_iter().enumerate().take(3) {
+            assert_binary_snapshot!(format!("page_{}.BIN", index + 1).as_str(), page_bytes);
```

Comment:
I wonder if we can make use of encoding images as `bmp` here, as that process is lossless it should be fine, but it'd allow us to actually look at the images during review, which could be neat 😊

The change _should_ be relatively straight forward and I think could look like this (rough sketch) - encoding via the `image` crate isn't always super straightforward, so thought this little snippet may help^^

```rs
let mut buffer = Vec::new();
let encoder = image::codecs::bmp::BmpEncoder::new(&mut buffer);

page
    .write_with_encoder(encoder)
    .expect("image should be able to be encoded into a bitmap");

assert_binary_snapshot!(&format!("page_{}.bmp", index + 1), buffer);
```

Reply:
Seeing the files does make comparing much easier! Just implemented this on my last commit.